app/components/Sidebar/components/DocumentLink.js

```diff
@@ -101,50 +108,93 @@ function DocumentLink({
-        <SidebarLink
-          innerRef={isActiveDocument ? activeDocumentRef : undefined}
-          onMouseEnter={handleMouseEnter}
-          to={{
-            pathname: node.url,
-            state: { title: node.title },
+    <>
+      <div
+        key={node.id}
+        ref={drag}
+        style={{
+          opacity: isDragging ? 0.6 : 1,
+          outline: isDragging ? "1px solid green" : "",
+        }}
+      >
+        <div
+          ref={drop}
+          style={{
+            outline: isOver && canDrop ? "1px solid red" : "",
           }}
-          label={
-            <>
-              {hasChildDocuments && (
-                <Disclosure
-                  expanded={expanded}
-                  onClick={handleDisclosureClick}
-                />
-              )}
-              <EditableTitle
-                title={node.title || t("Untitled")}
-                onSubmit={handleTitleChange}
-                canUpdate={canUpdate}
-              />
-            </>
-          }
-          depth={depth}
-          exact={false}
-          menuOpen={menuOpen}
-          menu={
-            document ? (
-              <Fade>
-                <DocumentMenu
-                  position="right"
-                  document={document}
-                  onOpen={() => setMenuOpen(true)}
-                  onClose={() => setMenuOpen(false)}
-                />
-              </Fade>
-            ) : undefined
-          }
-        ></SidebarLink>
-      </DropToImport>
-
-      {expanded && (
+        >
+          <DropToImport documentId={node.id} activeClassName="activeDropZone">
+            <SidebarLink
+              innerRef={isActiveDocument ? activeDocumentRef : undefined}
+              onMouseEnter={handleMouseEnter}
+              to={{
+                pathname: node.url,
+                state: { title: node.title },
+              }}
+              label={
+                <>
+                  {hasChildDocuments && (
+                    <Disclosure
+                      expanded={expanded && !isDragging}
+                      onClick={handleDisclosureClick}
+                    />
+                  )}
+                  <EditableTitle
+                    title={node.title || t("Untitled")}
+                    onSubmit={handleTitleChange}
+                    canUpdate={canUpdate}
+                  />
+                </>
+              }
+              depth={depth}
+              exact={false}
+              menuOpen={menuOpen}
+              menu={
+                document ? (
+                  <Fade>
+                    <DocumentMenu
+                      position="right"
+                      document={document}
+                      onOpen={() => setMenuOpen(true)}
+                      onClose={() => setMenuOpen(false)}
+                    />
+                  </Fade>
+                ) : undefined
+              }
+            />
+          </DropToImport>
+        </div>{" "}
+      </div>
```

Comment:
closed the draggable element right here, so that's the object that gets captured as the default drag layer, seems straightforward without having to deal with the custom drag layer stuff. I'm fine either way; maybe start with this and do the custom thing later if we need it?